apps/gp2-server/test/handlers/event/update-events-meeting-link-handler.test.ts

```diff
@@ -0,0 +1,200 @@
+import { when } from 'jest-when';
+
+import {
+  getCommonGoogleId,
+  meetingLinkUpdateHandler,
+} from '../../../src/handlers/event/update-events-meeting-link-handler';
+import { getEventEvent, getEventResponse } from '../../fixtures/event.fixtures';
+import { eventControllerMock } from '../../mocks/event.controller.mock';
+import { loggerMock } from '../../mocks/logger.mock';
+
+describe('getCommonGoogleId', () => {
+  test('works when it receives a series googleId event', () => {
+    expect(
+      getCommonGoogleId('65h0vfns41nlt7ra26los68kl2_20230920T133000Z'),
+    ).toEqual('65h0vfns41nlt7ra26los68kl2');
+  });
+
+  test('works when it receives a non series googleId event', () => {
+    expect(getCommonGoogleId('65h0vfns41nlt7ra26los68kl2')).toEqual(
+      '65h0vfns41nlt7ra26los68kl2',
+    );
+  });
+
+  test('works when it receives an empty string', () => {
+    expect(getCommonGoogleId('')).toEqual('');
+  });
+});
+
+describe('Meeting Link Update Handler', () => {
+  const handler = meetingLinkUpdateHandler(eventControllerMock, loggerMock);
+
+  afterEach(() => jest.clearAllMocks());
+
+  test('updates series events when there are events with common google ids', async () => {
+    const seriesEvent1 = getEventResponse();
+    seriesEvent1.id = 'event-series-1';
+    seriesEvent1.googleId = 'googleId_1';
+    seriesEvent1.meetingLink = 'https://zoom.com/shared-link';
+    seriesEvent1.copyMeetingLink = true;
+
+    const seriesEvent2 = getEventResponse();
+    seriesEvent2.id = 'event-series-2';
+    seriesEvent2.googleId = 'googleId_2';
+    seriesEvent2.meetingLink = undefined;
+
+    const seriesEvent3 = getEventResponse();
+    seriesEvent3.id = 'event-series-3';
+    seriesEvent3.googleId = 'googleId_3';
+    seriesEvent3.meetingLink = undefined;
+
+    eventControllerMock.fetchById.mockResolvedValueOnce(seriesEvent1);
+    eventControllerMock.fetch.mockResolvedValueOnce({
+      total: 3,
+      items: [seriesEvent1, seriesEvent2, seriesEvent3],
+    });
+
+    const webhookEvent = getEventEvent(seriesEvent1.id, 'EventsPublished');
+
+    await handler(webhookEvent);
+
+    expect(eventControllerMock.fetchById).toHaveBeenCalledWith(seriesEvent1.id);
+    expect(eventControllerMock.fetch).toHaveBeenCalledWith({
+      filter: { googleId: 'googleId' },
+    });
+
+    expect(eventControllerMock.update).toHaveBeenNthCalledWith(
```

Comment:
maybe you want to show that it has not been called with `seriesEvent1` data? i think otherwise it would pass even without filtering that event out, or maybe not because of the use of `nthCalled` but still

Reply:
in fact i dont think you need the nth-called expectations right? what difference does it make which order they are being updated in, for all i care they could be updated in parallel, as long as we skip the one that caused the change